R/table-subscript.R

```diff
@@ -641,6 +649,20 @@ addArrayIndicesIfMissing <- function(q.test.info, y, dim.names, qtypes, sep = "_
     return(cbind(arr.idx, q.test.info))
 }
 
+checkArrayIndices <- function(dim.names, q.test.indices)
+{
+    for (qname in colnames(q.test.indices)) {
+        dim.ind <- which(names(dim.names) == qname)
+        if (length(dim.ind) != 1)
+            next
+        if (any(!q.test.indices[,qname] %in% dim.names[[dim.ind]]))
+        {
+            return(FALSE)
+        }
+    }
+    return(TRUE)
```

Comment:
Could reduce this to the following?
```suggestion
    saved.qstat.dimnames <- q.test.indices |> lapply(levels)
    if (!identical(names(saved.qstat.dimnames), names(dim.names))) return(FALSE)
    differences <- mapply(setdiff, saved.dimnames, dim.names, SIMPLIFY = FALSE)
    all(lengths(differences) == 0L)
```

Reply:
Thanks. I guess the order of the columns can be assumed because of the way the indices were created? Fixed in [a07519f](https://github.com/Displayr/verbs/pull/121/commits/a07519f722101fdcbc453111a2a9f6aa4ddf6638)